test/e2e/mcd_test.go

```diff
@@ -143,6 +145,106 @@ func TestMCDeployed(t *testing.T) {
 	}
 }
 
+func TestUpdateSSH(t *testing.T) {
+	cs := framework.NewClientSet("")
+
+	// create a dummy MC with an sshKey for user Core
+	mcName := fmt.Sprintf("sshkeys-worker-%s", uuid.NewUUID())
+	mcadd := &mcv1.MachineConfig{}
+	mcadd.ObjectMeta = metav1.ObjectMeta{
+		Name:   mcName,
+		Labels: mcLabelForWorkers(),
+	}
+	// create a new MC that adds a valid user & ssh keys
+	tempUser := ignv2_2types.PasswdUser{Name: "core", SSHAuthorizedKeys: []ignv2_2types.SSHAuthorizedKey{"1234_test", "abc_test"}}
+	mcadd.Spec = mcv1.MachineConfigSpec{
+		Config: ignv2_2types.Config{
+			Ignition: ignv2_2types.Ignition{
+				Version: "2.2.0",
+			},
+			Passwd: ignv2_2types.Passwd{
+				Users: []ignv2_2types.PasswdUser{tempUser},
+			},
+		},
+	}
+	_, err := cs.MachineConfigs().Create(mcadd)
+	if err != nil {
+		t.Errorf("failed to create machine config %v", err)
+	}
+
+	// grab the latest worker- MC
+	var newMCName string
+	if err := wait.PollImmediate(2*time.Second, 5*time.Minute, func() (bool, error) {
+		mcp, err := cs.MachineConfigPools().Get("worker", metav1.GetOptions{})
+		if err != nil {
+			return false, err
+		}
+		for _, mc := range mcp.Status.Configuration.Source {
+			if mc.Name == mcName {
+				newMCName = mcp.Status.Configuration.Name
+				return true, nil
+			}
+		}
+		return false, nil
+	}); err != nil {
+		t.Errorf("machine config hasn't been picked by the pool: %v", err)
+	}
+
+	visited := make(map[string]bool)
+	if err := wait.Poll(2*time.Second, 5*time.Minute, func() (bool, error) {
+		nodes, err := getNodesByRole(cs, "worker")
+		if err != nil {
+			return false, err
+		}
+		for _, node := range nodes {
+			if visited[node.Name] {
+				continue
+			}
+			// check that the new MC is in the annotations and that the MCD state is Done.
+			if node.Annotations[constants.CurrentMachineConfigAnnotationKey] == newMCName &&
+				node.Annotations[constants.MachineConfigDaemonStateAnnotationKey] == constants.MachineConfigDaemonStateDone {
+				// oc get pods -n openshift-machine-config-operator --field-selector spec.nodeNAME=ip-10-0-134-90.us-west-2.compute.internal
+				// oc rsh -n openshift-machine-config-operator machine-config-daemon-zt85f
+				// grep -c  1234_test /rootfs/home/core/.ssh/authorized_keys
+
+				// find the MCD pod that has spec.nodeNAME = node.Name and get its name:
+				listOptions := metav1.ListOptions{
+					FieldSelector: fields.SelectorFromSet(fields.Set{"spec.nodeName": node.Name}).String(),
+				}
+				listOptions.LabelSelector = labels.SelectorFromSet(labels.Set{"k8s-app": "machine-config-daemon"}).String()
+
+				mcdList, err := cs.Pods("openshift-machine-config-operator").List(listOptions)
+				if err != nil {
+					t.Fatalf("%#v", err)
+				}
+				mcdName := mcdList.Items[0].ObjectMeta.Name
```

Comment:
check `len(mcdList)` or this will just panic

Reply:
done!!